modules/tinymce/src/core/test/ts/browser/fmt/FormatEmptyLineTest.ts

```diff
@@ -0,0 +1,125 @@
+import { Log, Pipeline, Step } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { Obj } from '@ephox/katamari';
+import { TinyApis, TinyLoader, TinyUi } from '@ephox/mcagar';
+import Editor from 'tinymce/core/api/Editor';
+import SilverTheme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.core.fmt.FormatEmptyLineTest', (success, failure) => {
+  SilverTheme();
+
+  TinyLoader.setupLight((editor: Editor, onSuccess, onFailure) => {
+    const tinyApis = TinyApis(editor);
+    const tinyUi = TinyUi(editor);
+
+    const tagHTML = (tag: string) => `<${tag}>a</${tag}><${tag}>&nbsp;</${tag}><${tag}>b</${tag}>`;
+
+    const sToggleInlineStyle = (style: string) => tinyUi.sClickOnToolbar(`click ${style}`, `[aria-label="${style}"]`);
+    const sApplyCustomFormat = (format: string) => Step.sync(() => editor.formatter.apply(format));
+    const sRemoveCustomFormat = (format: string) => Step.sync(() => editor.formatter.remove(format));
+
+    const sSelectAll = Step.sync(() => editor.execCommand('SelectAll'));
+
+    const tableHTML = `<table style="border-collapse: collapse; width: 100%;" border="1">
+    <tbody>
+    <tr>
+    <th scope="row">&nbsp;</th>
+    </tr>
+    <tr>
+    <td>b</td>
+    </tr>
+    <tr>
+    <td>&nbsp;</td>
+    </tr>
+    </tbody>
+    </table>`;
+
+    const listHTML = `<ul>
+    <li>b</li>
+    <li>&nbsp;</li>
+    </ul>`;
+
+    const sTestFormat = (label: string, styleSelector: string, selectorCount: number, html: string, sSelect: Step<any, any>, sApplyFormat: Step<any, any>, sRemoveFormat: Step<any, any>) => {
+      const expectedPresence: Record<string, number> = {};
+      expectedPresence[styleSelector] = selectorCount;
+      const expectedPresenceOnRemove = Obj.map(expectedPresence, (val, key) => key === styleSelector ? 0 : val);
```

Comment:
Could this all be simplified? Something like this?
```typescript
const expectedPresence = { [styleSelector]: selectorCount };
const expectedPresenceOnRemove = { [styleSelector]: 0 };
```

Reply:
Thanks for suggestion. Updated.